tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -0,0 +1,131 @@
+import math
+import pickle
+
+import boto3
+
+from tron.serialize.runstate.shelvestore import ShelveKey
+
+OBJECT_SIZE = 400000
+
+
+class DynamoDBStateStore(object):
+    def __init__(self, name, dynamodb_region) -> None:
+        self.dynamodb = boto3.resource('dynamodb', region_name=dynamodb_region)
+        self.client = boto3.client('dynamodb', region_name=dynamodb_region)
+        self.name = name
+        self.table = self.dynamodb.Table(name.replace('/', '-'))
+
+    def build_key(self, type, iden) -> ShelveKey:
+        """
+        It builds a unique partition key. The key could be objects with __str__ method.
+        """
+        return f"{type} {iden}"
+
+    def restore(self, keys) -> dict:
+        """
+        Fetch all under the same parition key(keys).
+        ret: <dict of key to states>
+        """
+        items = zip(
+            keys,
+            (self[key] for key in keys),
+        )
+
+        return {k: v for k, v in items if v}
+
+    def alert(self):
```

Comment:
move to mirror store

Reply:
I'm thinking about keeping alert here even after mirrorstore is removed.

Reply:
we're only alerting on inconsistency between shelve and dynamodb so it only makes sense to have this function in mirror store

Reply:
In this case, yes. I mean I'm thinking about adding checks for dynamodb. So I keep it here for the future.